contracts/stellar-interchain-token-service/src/migrate.rs

```diff
@@ -0,0 +1,138 @@
+use soroban_sdk::{Address, BytesN, Env, Vec};
+use stellar_axelar_std::interfaces::{CustomMigratableInterface, UpgradableClient};
+
+use crate::error::ContractError;
+use crate::{deployer, storage, InterchainTokenService};
+
+pub mod legacy_storage {
+    use soroban_sdk::{contracttype, Address, BytesN, String};
+    use stellar_axelar_std::contractstorage;
+
+    use crate::storage::TokenIdConfigValue;
+
+    #[contractstorage]
+    enum DataKey {
+        #[instance]
+        #[value(Address)]
+        Gateway,
+
+        #[instance]
+        #[value(Address)]
+        GasService,
+
+        #[instance]
+        #[value(String)]
+        ChainName,
```

Comment:
shall we keep only the variants that are used around? pro: more readable, better coverage. cons: not just a copy paste of the data key, accidental edits